arangod/Cluster/CreateCollection.cpp

```diff
@@ -273,9 +273,12 @@ bool CreateCollection::createReplication2Shard(CollectionID const& collection,
         replication2::replicated_state::document::DocumentLeaderState>(
         state.get()->getLeader());
     if (leaderState != nullptr) {
-      leaderState->createShard(
-          shard, collection,
-          velocypack::SharedSlice(_description.properties()->bufferRef()));
+      // TODO we are blocking the maintenance here, but we should not
+      leaderState
+          ->createShard(
+              shard, collection,
+              velocypack::SharedSlice(_description.properties()->bufferRef()))
+          .get();
```

Comment:
We can simply put shard creation on the slow lane. This slow lane is designed for executing long running processes like synchronize shard (which takes days). We can wait for a shard log entry to be replicated.

Reply:
you do by calling `get` on a future?